File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* mergeTwoLists(struct ListNode* list1, struct ListNode* list2){
    struct ListNode *temp1 = list1;
    struct ListNode *temp2 = list2;
    struct ListNode *list3 = NULL;
        if(temp1 == NULL && temp2 != NULL){
            return temp2;
        }else if(temp1 != NULL && temp2 == NULL){
            return temp1; 
        }else if(temp1 == NULL && temp2 == NULL){
            return NULL;
        }
        if(temp1 -> val <= temp2 -> val){
           list3 = temp1;
           list3 -> next = mergeTwoLists(temp1->next, temp2);
        }else if(temp1 -> val > temp2 -> val){
            list3 = temp2;
            list3 -> next = mergeTwoLists(temp1, temp2->next);
        }else{
            return NULL;
        }
    
    return list3;
}

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        curr_node = ListNode() 
        temp = curr_node
        while list1 and list2:
            if list1.val < list2.val:
                curr_node.next = ListNode(list1.val)
                curr_node = curr_node.next
                list1 = list1.next
            else:
                curr_node.next = ListNode(list2.val)
                curr_node = curr_node.next
                list2 = list2.next
        if list1:
            curr_node.next = list1
        if list2:
            curr_node.next = list2
        return temp.next
        